src/backend/distributed/planner/distributed_planner.c

```diff
@@ -527,6 +527,30 @@ GetRTEIdentity(RangeTblEntry *rte)
 }
 
 
+/*
+ * GetQueryLockMode returns the necessary lock mode to be acquired for the
+ * given query. (See comment written in RangeTblEntry->rellockmode)
+ */
+LOCKMODE
+GetQueryLockMode(Query *query)
+{
+	Assert(query != NULL && IsA(query, Query));
```

Comment:
When can a query type not be a query?

Reply:
It can't. It is just to ensure that function is called for a `Query` object.

Reply:
Seems not super useful.